contracts/test/unit/oracle/L2ScalingPriceOracle.t.sol

```diff
@@ -0,0 +1,500 @@
+    }
+
+    function testDeploymentSucceedsWithCorrectStartingOraclePrice() public {
+        l2scalingPriceOracle = new MockL2ScalingPriceOracle(
+            oracle,
+            jobId,
+            fee,
+            currentMonth,
+            previousMonth,
+            block.timestamp,
+            2e18
+        );
+        assertEq(l2scalingPriceOracle.oraclePrice(), 2e18);
+    }
+
+    function testFulfillSucceedsTwentyPercentTwelveMonthsStartingOraclePriceTwo()
+        public
+    {
+        l2scalingPriceOracle = new MockL2ScalingPriceOracle(
+            oracle,
+            jobId,
+            fee,
+            currentMonth,
+            previousMonth,
+            block.timestamp,
+            2e18
+        );
+        assertEq(l2scalingPriceOracle.oraclePrice(), 2e18);
+        vm.warp(block.timestamp + 33 days);
+        uint256 newCurrentMonth = (currentMonth * 120) / 100;
+        uint256 price = l2scalingPriceOracle.getCurrentOraclePrice();
+
+        for (uint256 i = 0; i < 12; i++) {
+            uint256 storedCurrentMonth = l2scalingPriceOracle.currentMonth();
+
+            vm.warp(block.timestamp + 31 days);
+            bytes32 requestId = l2scalingPriceOracle.requestCPIData();
+            /// prank to allow request to be fulfilled
+            vm.prank(address(0));
+            /// this will succeed as max allowable is 20%
+            l2scalingPriceOracle.fulfill(requestId, newCurrentMonth);
+            assertEq(l2scalingPriceOracle.getCurrentOraclePrice(), price);
+
+            uint256 expectedChangeRateBasisPoints = ((l2scalingPriceOracle
+                .currentMonth() - l2scalingPriceOracle.previousMonth()) *
+                10_000) / l2scalingPriceOracle.previousMonth();
+
+            assertEq(
+                l2scalingPriceOracle.monthlyChangeRateBasisPoints(),
+                expectedChangeRateBasisPoints.toInt256()
+            );
+            assertEq(l2scalingPriceOracle.previousMonth(), storedCurrentMonth);
+            assertEq(l2scalingPriceOracle.currentMonth(), newCurrentMonth);
+
+            assertEq(l2scalingPriceOracle.oraclePrice(), price);
+
+            newCurrentMonth =
+                (newCurrentMonth * (10_000 + expectedChangeRateBasisPoints)) /
+                10_000;
+            price = (price * (10_000 + expectedChangeRateBasisPoints)) / 10_000;
+        }
+    }
+
+    function testFulfillSucceedsTwentyPercentTwelveMonthsFuzz(uint128 x)
+        public
+    {
+        /// instead of using vm.assume, do this to cut down on wasted runs
+        if (x < 1e18) {
+            vm.expectRevert(
+                bytes("L2ScalingPriceOracle: Starting oracle price too low")
+            );
+        }
+
+        l2scalingPriceOracle = new MockL2ScalingPriceOracle(
+            oracle,
+            jobId,
+            fee,
+            currentMonth,
+            previousMonth,
+            block.timestamp,
+            x
+        );
+        if (x < 1e18) {
+            return;
+        }
```

Comment:
In the case of, $x<1e18$, wouldn't this test revert anyways, so do we still have to have to check the condition again to have the function exit early as well?

Reply:
I don't think that the test will automatically end

Reply:
removed the return, and the test fails because expectRevert doesn't exit the test after the next contract deployment